### ISSUE TITLE
sequelize db:migrate fails on localhost without internet connection

### ISSUE THREAD

**Issue by matt222r** [sequelize/sequelize#7530]
On my win10 laptop, in airplane mode:
```shell
> sequelize db:migrate

Sequelize [Node: 6.9.2, CLI: 2.7.0, ORM: 3.30.4]

Loaded configuration file "config\config.json".
Unable to connect to database: SequelizeConnectionError: getaddrinfo ENOENT localhost:5432
```

Works fine as soon as the WiFi reconnects.

My config.json:
```json
{
	"username": "postgres",
	"password": "password",
	"database": "whatever",
	"host": "localhost",
	"dialect": "postgres"
}
```

**Comment by sushantdhiman** [comment]
I think there is no postgres server running at `localhost:5432`

**Comment by matt222r** [comment]
That's incorrect; as soon as wi-fi is connected, the connection works, with the same config file (which points at localhost). I'm quite certain the relevant database is hosted locally.

**Comment by sushantdhiman** [comment]
@matt222r Can you show me `ping` to `localhost:5432` after turning off wifi (screenshot ?)

**Comment by matt222r** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/27628301/25539737/35893836-2c0e-11e7-8f5e-f30d37ce01b0.jpg)

One should not be able to literally ping localhost:5432; ping isn't a port-specific utility. localhost, of course, is still available.

**Comment by sushantdhiman** [comment]
@matt222r Right, we cant use ping here. I just wanted you to confirm if Postgres server is available after you disconnect wifi, because we dont have any code which says system should be connected to internet.

This indicate there can be only but one issue that somehow wifi is taking down your Postgres server. Please confirm ?

**Comment by matt222r** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/27628301/25540671/92b1c610-2c11-11e7-89f4-c39c22df1890.jpg)

No, psql works fine with wifi off. It's using a socket over localhost, as verified with Resource Monitor. I tried it over IPv4 and IPv6; both work.

Perhaps this is the result of doing a DNS lookup on "localhost"?

**Comment by sushantdhiman** [comment]
If you are using socket file you can pass it connection string as `host` https://github.com/brianc/node-postgres/wiki/Client#new-clientstring-domainsocketfolder-client

**Comment by sushantdhiman** [comment]
Also have you tried `127.0.0.1`  instead of localhost, if that made any difference ?

**Comment by matt222r** [comment]
If I replace "localhost" in the config file with "127.0.0.1", the problem does not occur.

**Comment by dmmulroy** [comment]
I also am running into this issue.

**Comment by chanphillip** [comment]
I had this issue and fixed it by replacing with `127.0.0.1`

**Comment by AmitBenAmi** [comment]
I also had this issue, and it was fixed by replacing to `127.0.0.1`
